src/array/schema.rs

```diff
@@ -20,41 +21,88 @@ impl ArrayType {
 
 /// Wrapper for the CAPI handle.
 /// Ensures that the CAPI structure is freed.
-pub(crate) struct RawSchema {
-    ffi: *mut ffi::tiledb_array_schema_t,
-}
-
-impl RawSchema {
-    pub fn new(ffi: *mut ffi::tiledb_array_schema_t) -> Self {
-        RawSchema { ffi }
-    }
+pub(crate) enum RawSchema {
+    Owned(*mut ffi::tiledb_array_schema_t),
 }
 
 impl Deref for RawSchema {
     type Target = *mut ffi::tiledb_array_schema_t;
 
     fn deref(&self) -> &Self::Target {
-        &self.ffi
+        let RawSchema::Owned(ref ffi) = *self;
+        ffi
     }
 }
 
 impl Drop for RawSchema {
     fn drop(&mut self) {
-        unsafe { ffi::tiledb_array_schema_free(&mut self.ffi) }
+        unsafe {
+            let RawSchema::Owned(ref mut ffi) = *self;
+            ffi::tiledb_array_schema_free(ffi)
+        }
     }
 }
 
 pub struct Schema<'ctx> {
     context: &'ctx Context,
     raw: RawSchema,
-    _domain: Domain<'ctx>,
+    domain: Domain<'ctx>,
```

Comment:
Same thing here as for the Domain.dimensions vector. We don't want to hold state here as that means we're doing a shadow state/cache thing which is what fuels my nightmares.